gtsfm/common/gtsfm_data.py

```diff
@@ -0,0 +1,196 @@
+        self._tracks: List[SfmTrack] = []
+        self._number_images = number_images
+
+    def __eq__(self, other: object) -> bool:
+        """Checks equality with the other object."""
+
+        if not isinstance(other, GtsfmData):
+            return False
+
+        if self._number_images != other.number_images():
+            return False
+
+        for i, cam in self._cameras.items():
+            other_cam = other.get_camera(i)
+            if not cam.equals(other_cam, EQUALITY_TOLERANCE):
+                return False
+
+        for k in range(self.number_tracks()):
+            track = self.get_track(k)
+            other_track = other.get_track(k)
+
+            if track.number_measurements() != other_track.number_measurements():
+                return False
+
+            for j in range(track.number_measurements()):
+                i, uv = track.measurement(j)
+                other_i, other_uv = other_track.measurement(j)
+
+                if i != other_i:
+                    return False
+                if not np.allclose(uv, other_uv):
+                    return False
+
+        return True
+
+    def number_images(self) -> int:
+        """Getter for the number of images.
+
+        Returns:
+            Number of images.
+        """
+        return self._number_images
+
+    def number_tracks(self) -> int:
+        """Getter for the number of tracks.
+
+        Returns:
+            Number of tracks.
+        """
+        return len(self._tracks)
+
+    def get_valid_camera_indices(self) -> List[int]:
+        """Getter for image indices where there is a valid (not None) camera.
+
+        Returns:
+            List of indices with a valid camera.
+        """
+        return list(self._cameras.keys())
+
+    def get_camera(self, index: int) -> Optional[PinholeCameraCal3Bundler]:
+        """Getter for camera.
+
+        Args:
+            index: the image index to fetch the camera for.
+
+        Returns:
+            The camera if it is a valid one, None otherwise.
+        """
+        return self._cameras.get(index)
+
+    def get_track(self, index: int) -> SfmTrack:
+        """Getter for the track.
+
+        Args:
+            index: track index to fetch.
+
+        Returns:
+            Requested track.
+        """
+        return self._tracks[index]
+
+    def add_track(self, track: SfmTrack) -> bool:
+        """Add a track, after checking if all the cameras in the track are already added.
+
+        Args:
+            track: track to add.
+
+        Returns:
+            Flag indicating the success of adding operation.
+        """
+        # check if all cameras are already added
+        for j in range(track.number_measurements()):
+            i, _ = track.measurement(j)
+
+            if i not in self._cameras:
+                return False
+
+        self._tracks.append(track)
+        return True
+
+    def add_camera(self, camera: PinholeCameraCal3Bundler, camera_index: int) -> None:
```

Comment:
this is a very minor nitpick, but i think having args in order (self, camera_index, camera) might make mores sense, since it symbolizes to me what this does:
`dict[cam_idx] = cam`

Reply:
thanks